elevation_layer/src/elevation_layer.cpp

```diff
@@ -0,0 +1,242 @@
+//
+// Created by eugenio on 16.10.18.
+//
+
+#include <elevation_layer.h>
+#include <pluginlib/class_list_macros.h>
+
+PLUGINLIB_EXPORT_CLASS(elevation_layer::ElevationLayer, costmap_2d::Layer)
+
+using costmap_2d::NO_INFORMATION;
+using costmap_2d::LETHAL_OBSTACLE;
+using costmap_2d::FREE_SPACE;
+
+using costmap_2d::ObservationBuffer;
+using costmap_2d::Observation;
+
+namespace elevation_layer
+{
+
+    ElevationLayer::ElevationLayer() : filterChain_("grid_map::GridMap")
+    {
+        costmap_ = NULL;  // this is the unsigned char* member of parent class Costmap2D.
+    }
+
+    void ElevationLayer::onInitialize() {
+        ros::NodeHandle nh("~/" + name_), g_nh;
+        rolling_window_ = layered_costmap_->isRolling();
+
+        ElevationLayer::matchSize();
+        current_ = true;
+        elevation_map_received_ = false;
+        filters_configuration_loaded_ = false;
+        global_frame_ = layered_costmap_->getGlobalFrameID();
```

Comment:
`global_frame_` is never used in the code!

Reply:
AFAIK, the elevation layer can only work if the costmap frame is set to `odom` (or same as the frame used by elevation mapping). The user should be warned if the frames in costmap and elevation layers are not matching.